vfx/gl/texture.cc

```diff
@@ -0,0 +1,81 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "vfx/gl/texture.h"
+
+#include "ui/gl/gl_bindings.h"
+
+namespace vfx {
+
+Texture::Desc::Desc()
+  : internal_format(GL_RGBA),
+    format(GL_RGBA),
+    type(GL_UNSIGNED_BYTE) { 
+}
+
+Texture::Texture() : id_(0) {
+}
+
+Texture::Texture(const Desc& desc) : id_(0), size_(desc.size) {
+  glGenTextures(1, &id_);
+  glBindTexture(GL_TEXTURE_2D, id_);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glTexImage2D(GL_TEXTURE_2D, 0, desc.internal_format, desc.size.width(),
```

Comment:
I don't know if your OpenGL library prints errors automagically (if not you should probably get yourself [a macro that checks GL errors](http://stackoverflow.com/questions/11256470/define-a-macro-to-facilitate-opengl-command-debugging) and wrap all your GL calls in it), but TexImage2D has approximately a billion things that can go wrong with it, so it could be worth manually checking here.